### DISCUSSION TITLE
Identify if the Hinode theme is used

### DISCUSSION THREAD

**Discussion by myrthos** [gethinode/hinode#470] [Help] [answered]
For a module, the template project is not using the Hinode theme.  
I would like to know if my shortcode is being run in the Hinode theme or not. In case it is not, I would like to add some extra styling in the shortcode.  
Also, if someone wants to use the shortcode with a non-Hinode theme, it would still work, as at the moment it doesn't.

Is there currently something to identify if we re using the Hinode theme or not? If not, would you be willing to add such an identifier?

**Comment by markdumay** [comment]
I don't think there's an out-of-the-box option provided by Hugo to detect the current theme / included modules. Then again, a shortcode has access to both page variables and site variables. We could consider a new site parameter, such as `params.main.theme = "hinode"`. 

Another option is to test for the availability of a specific file. For example, if we could read a file `_vendor/github.com/gethinode/hinode/theme.toml`, we could assume Hinode is being used. It does require vendoring of all modules. I don't believe we have access to the local module cache folder from within a shortcode.

**Comment by myrthos** [comment, marked as answer]
I thought about a parameter, but then it would only work from a certain version of Hinode onwards. If someone would have an older version of Hinode the shortcode would detect Hinode is not present. Presence of a file can also be tricky.  
Currently I have it as a shortcode parameter and leave it to the user to modify it, if that is required. At the moment it is the only deterministic solution I can think of.

**Reply by markdumay** [reply]
This seems the best solution for now. Indeed, the other two approaches cannot be guaranteed.

**Comment by markdumay** [comment]
What comes to mind, you could also check for the existence of `site.Params.modules`. This is the configuration used by Hinode to determine if a component is `core` or `optional`.
